### PR TITLE
scala3.2: new port (version 3.2.0)

### PR DESCRIPTION
**Pull request by catap**
#### Description

Scala-3.2.0 is released.

I've used this opportunity to heavy reworked scalas Portfiles.

Instead of a dozen of scalaX.Y, I've migrated everything to one Portfile.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.5.1 21G83 x86_64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
Cc: @blair

**Comment by catap** [comment]
@blair I've installed all ports and confirm that layout is right. Select is working and scala and scalac can be used.

Thus, I've rebased it to the last master with hope that CI will pass because I haven o idea why did it fail.

File: lang/scala-docs/Portfile
Comment (mascguy) [inline lang/scala-docs/Portfile:15]: No need for `platforms darwin`, as this is now the default.

Comment (catap) [reply lang/scala-docs/Portfile]: Fixed

File: sysutils/scala_select/Portfile
Comment (mascguy) [inline sysutils/scala_select/Portfile:23]: You can simplify this port, via use of pg `stub`:
* Include that first, followed by pg `select`.
* Change phase `destroot` to `post-destroot`.
* Eliminate `platforms`, `license`, `maintainers`, and `homepage`.
* Eliminate `use_configure`, as well as phase `build`.

Comment (mascguy) [reply sysutils/scala_select/Portfile:23]: Also be sure to run `port lint --nitpick` on all of your ports, to catch issues like the following:

```
port lint scala_select:
Warning: Line 5 contains tab but modeline says tabs should be expanded
```

Comment (catap) [reply sysutils/scala_select/Portfile]: @mascguy thanks!

File: sysutils/scala_select/Portfile
Comment (mascguy) [inline sysutils/scala_select/Portfile:17]: You can eliminate `distfiles` too, as that's also cleared by pg `stub`.

Comment (catap) [reply sysutils/scala_select/Portfile]: @mascguy thanks!

**Comment by mascguy** [comment]
I just cleaned up a number of `xxx_select` ports, including `scala_select`. So you'll need to rebase against master.

**Comment by catap** [comment]
@mascguy its done

**Review by mascguy — APPROVED** [review]
LGTM, I'll merge once CI completes.

**Comment by catap** [comment]
@mascguy the fast question: if I would like to retire some ports. Shall I use `obsolete` PG or do something better?

The idea is retire scala-2.7, scala-2.8 which can't be run on MacPorts because it requires JDK up to 7.

**Comment by catap** [comment]
@mascguy yep, everything under 2.10 requires jdk-1.7. Thus, it is officially unmaintained :)

**Comment by catap** [comment]
@mascguy let start from mark it deprecated.

**Comment by mascguy** [comment]
> @mascguy let start from mark it deprecated. If someone had installed JDK1.7 — it works :)

Sounds good. But it looks like you're including pg `deprecated` for all ports, including non-deprecated ones...?

**Comment by catap** [comment]
@mascguy yes, but it hasn't change anything. To enable a warning I should trigger it. Like `deprecated.eol_version yes`.

**Comment by catap** [comment]
> That said, if it only comes into play when ports specify the `deprecated.xxx` options, then that's fine.

It also makes `port lint --nitpick` happy ;) otherwise it starts to complain about duplicated PG.

**Review by mascguy — APPROVED** [review]

**Pull request merged**
